2-mandatory/3-groceries.js

```diff
@@ -27,13 +27,27 @@ Exercise 1:
 */
 // Gather all week item names into this array
 let weeklyGroceriesToBuy = [];
-
+for (const key in weeklyMealPlan) {
+   weeklyMealPlan[key].forEach(ingredient => {
+     if (!weeklyGroceriesToBuy.includes(ingredient)) {
+       weeklyGroceriesToBuy.push(ingredient);
+     }
+   })
+  };
 /*
```

Comment:
Great use of for...in  statement and implementation of an if statement. I have used ...new Set and Object.values to sort the assignment. Both solutions work great.